docs/function-inlining.md

```diff
@@ -0,0 +1,40 @@
+# No support for user inlining (yet)
+
+## Summary
+<!-- https://luau-lang.org/performance#function-inlining-and-loop-unrolling -->
+Luau's bytecode compiler (or VM?) is smart about inlining functions. We would rather focus on improving this automation than risk the chance of its misuse.
+
+## Motivation
+
+**context**: function inlining replaces function calls with the function's implementation to reduce overhead costs.
+
+Whether the goal is to improve instruction locality or perform context specific optimizations, <!-- force the compiler to optimize the function body in specific contexts --> function inlining can be a great resource for developers. Thankfully with the `-02` switch, Luau is able to automatically decide if code deserves to be inlined and optimize the code accordingly. To the best of our knowledge, exposing this to users directly could do more harm than good. Not only do these use cases only cater to a tiny subset of the language's users, it exposes our bytecode compiler to FILL when the larger subset who may not have a working understanding of the compiler adopts it.
```

Comment:
I'm not sure I understand "it exposes our bytecode compiler to FILL when the larger subset who may not have a working understanding of the compiler adopts it."

Reply:
"FILL" is a placeholder